### DISCUSSION TITLE
Looking for source code samples

### DISCUSSION THREAD

**Discussion by MarcusE1W** [modula3/cm3#814] [Q&A] [answered]
Hi.
I just started with Modula 3 the other day and read a bit this and that. Now I am looking for some source code that I have found mentioned:

1. Monster
On page 4, exercise 1 of the [Tristle Tutorial](http://modula3.github.io/cm3/src_reports/src-069.pdf)  is a stub for a cascading view module. The main part of the recursion is missing. After staring at the stub and at the picture (Figure 2, page 5) to illustrate the result I have to say I have no idea how to create such brilliant chaos with a simple recursion. Is anyone aware of the solution for exercise 1 ?

2. Distributed Spreadsheet
There is an [article](https://www.linuxjournal.com/article/2690)  that explains some principals of how to develop a distributed spreadsheet based on the spreadsheet example shipped with TurboPascal. Does anyone have the source code for the Modula 3 example?

**Comment by mikanystrom** [comment, marked as answer]
The answer to 1 is on pp. 61-62 of the Trestle Tutorial.

**Comment by MarcusE1W** [comment]
Ah, I see. The chaotic design is done manually after execution.
Still it is astonishing that you can copy and past an example from 1994 and it compiles and runs on a aarch64 Pinebook Pro in 2022 without a hick...
